airflow/operators/python_operator.py

```diff
@@ -86,6 +87,38 @@ def __init__(
         if templates_exts:
             self.template_ext = templates_exts
 
+    def __deepcopy__(self, memo):
+        """
+        Hack sorting double chained task lists by task_id to avoid hitting
+        max_depth on deepcopy operations.
+        """
+        sys.setrecursionlimit(5000)  # TODO fix this in a better way
+        cls = self.__class__
+        result = cls.__new__(cls)
+        memo[id(self)] = result
+
+        # since we won't mutate the arguments, we should just do the shallow copy
+        # there are some cases we can't deepcopy the objects(e.g protobuf).
+        shallow_copy = {'python_callable', 'op_kwargs'}
```

Comment:
It could be nice to make this method part of BaseOperator and make `shallow_copy_args` a class attribute. If doing this, I'd push to have a `_base_operator_shallow_copy_args` class attribute that would get merged with `shallow_copy_args`. That would make it easy to make args shallow copied for any operator.

Reply:
thanks. This is better. pr updated.